modules/core/src/test/java/org/apache/ignite/failure/FailureHandlingConfigurationTest.java

```diff
@@ -105,6 +110,60 @@
         cleanPersistenceDir();
     }
 
+    /**
+     * @throws Exception If failed
+     */
+    @Test
+    public void testGetorCreateLongCacheNameExceed235() throws Exception{
+        IgniteEx ignite = startGrid(0);
+        ignite.cluster().active(true);
+        final CacheConfiguration cfg = new CacheConfiguration(DEFAULT_CACHE_NAME);
+        String cacheName = GridTestUtils.randomString(250);
+        cfg.setName(cacheName);
+        GridTestUtils.assertThrows(log, new Callable<Object>() {
+            @Override
+            public Object call() throws Exception {
+                return ignite.getOrCreateCache(cfg);
+            }
+        }, IllegalArgumentException.class, "Ouch! Argument is invalid: Length of cache name can not exceed 235.");
+    }
+
+    /**
+     * @throws Exception If failed
+     */
+    @Test
+    public void testGetorCreateLongCacheGroupNameExceed235() throws Exception{
```

Comment:
I suggest renaming the test to smth. like `testCreateLongCacheGroupName` and move it to `org.apache.ignite.internal.processors.cache.GridCacheConfigurationConsistencySelfTest` test class.